src/widget/tool/screenshotgrabber.cpp

```diff
@@ -0,0 +1,183 @@
+    
+    return QWidget::eventFilter(object, event);
+}
+
+void ScreenshotGrabber::showGrabber()
+{
+    this->screenGrab = grabScreen();
+    this->screenGrabDisplay->setPixmap(this->screenGrab);
+    this->window->show();
+    this->window->setFocus();
+    this->window->grabKeyboard();
+    adjustWindowSize();
+    adjustTooltipPosition();
+}
+
+bool ScreenshotGrabber::handleKeyPress(QKeyEvent* event)
+{
+    if (event->key() == Qt::Key_Escape)
+        reject();
+    else if (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter)
+        acceptRegion();
+    else
+        return false;
+    
+    return true;
+}
+
+void ScreenshotGrabber::acceptRegion()
+{
+    QRect rect = this->chooserRect->chosenRect();
+    if (rect.width() < 1 || rect.height() < 1)
+        return;
+    
+    // 
+    qDebug() << "Screenshot accepted, chosen region" << rect;
+    emit screenshotTaken(this->screenGrab.copy(rect));
+    this->window->close();
+}
+
+void ScreenshotGrabber::setupWindow()
+{
+    this->window->setWindowFlags(Qt::FramelessWindowHint | Qt::X11BypassWindowManagerHint);
+    this->window->setAttribute(Qt::WA_DeleteOnClose);
+    this->window->setContentsMargins(0, 0, 0, 0);
+    this->window->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    this->window->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    this->window->setFrameShape(QFrame::NoFrame);
+    
+    connect(this->window, &QObject::destroyed, this, &QObject::deleteLater);
+    this->window->installEventFilter(this);
+}
+
+void ScreenshotGrabber::setupScene(QGraphicsScene* scene)
+{
+    this->overlay = new ScreenGrabberOverlayItem(this);
+    this->helperToolbox = new ToolBoxGraphicsItem;
+    
+    this->screenGrabDisplay = scene->addPixmap(this->screenGrab);
+    this->helperTooltip = scene->addText(QString());
+    scene->addItem(this->overlay);
+    this->chooserRect = new ScreenGrabberChooserRectItem(scene);
+    scene->addItem(this->helperToolbox);
+    
+    this->helperToolbox->addToGroup(this->helperTooltip);
+    this->helperTooltip->setDefaultTextColor(Qt::black);
+    useNothingSelectedTooltip();
+    
+    connect(this->chooserRect, &ScreenGrabberChooserRectItem::doubleClicked, this, &ScreenshotGrabber::acceptRegion);
+    connect(this->chooserRect, &ScreenGrabberChooserRectItem::regionChosen, this, &ScreenshotGrabber::useRegionSelectedTooltip);
+    connect(this->chooserRect, &ScreenGrabberChooserRectItem::regionChosen, this->overlay, &ScreenGrabberOverlayItem::setChosenRect);
+}
+
+void ScreenshotGrabber::useNothingSelectedTooltip()
+{
+    this->helperTooltip->setHtml(tr("Click and drag to select a region to select it. Press <b>ESC</b> to cancel.",
+                                    "Help text shown when no region has been selected yet"));
+    adjustTooltipPosition();
+}
+
+void ScreenshotGrabber::useRegionSelectedTooltip()
+{
+    this->helperTooltip->setHtml(tr("Press <b>Return</b> to accept, or move and resize the selected region.",
```

Comment:
I don't quite like this text. Return is more commonly known as "Enter", "accept" is vague, there is no text about cancelling and the text implies you can't make a new selection altogether. What about changing it to this:

`Press <b>Enter</b> to send a screenshot of the selected region, press <b>Escape</b> to cancel, or select a new region.`